programs/uxd/src/instructions/credix_lp/rebalance_request_create_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,283 @@
+        ],
+        bump,
+        seeds::program = credix_client::ID,
+    )]
+    pub credix_withdraw_epoch: Account<'info, credix_client::WithdrawEpoch>,
+
+    /// #14
+    /// CHECK: initialized by credix program, not manipulated by us
+    #[account(mut)]
+    pub credix_withdraw_request: AccountInfo<'info>,
+
+    /// #15
+    pub system_program: Program<'info, System>,
+    /// #16
+    pub credix_program: Program<'info, credix_client::program::Credix>,
+}
+
+pub(crate) fn handler(ctx: Context<RebalanceRequestCreateFromCredixLpDepository>) -> Result<()> {
+    let credix_global_market_state = ctx.accounts.depository.load()?.credix_global_market_state;
+    let collateral_mint = ctx.accounts.depository.load()?.collateral_mint;
+    let depository_pda_signer: &[&[&[u8]]] = &[&[
+        CREDIX_LP_DEPOSITORY_NAMESPACE,
+        credix_global_market_state.as_ref(),
+        collateral_mint.as_ref(),
+        &[ctx.accounts.depository.load()?.bump],
+    ]];
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Check if the withdraw request period is active at the moment
+    // ---------------------------------------------------------------------
+
+    let start_of_request_phase_timestamp = ctx.accounts.credix_withdraw_epoch.go_live;
+    let end_of_request_phase_timestamp = start_of_request_phase_timestamp
+        .checked_add(ctx.accounts.credix_withdraw_epoch.request_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    require!(
+        (start_of_request_phase_timestamp..end_of_request_phase_timestamp)
+            .contains(&Clock::get()?.unix_timestamp),
+        UxdError::InvalidCredixWithdrawEpochRequestPeriod,
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Compute the profits and overflow amounts of the depository
+    // ---------------------------------------------------------------------
+
+    let redeemable_amount_under_management = checked_convert_u128_to_u64(
+        ctx.accounts
+            .depository
+            .load()?
+            .redeemable_amount_under_management,
+    )?;
+
+    let overflow_value = {
+        let redeemable_amount_under_management_target_amount =
+            calculate_credix_lp_depository_target_amount(
+                &ctx.accounts.controller,
+                &ctx.accounts.identity_depository,
+                &ctx.accounts.mercurial_vault_depository,
+                &ctx.accounts.depository,
+            )?;
+        if redeemable_amount_under_management < redeemable_amount_under_management_target_amount {
+            0
+        } else {
+            redeemable_amount_under_management
+                .checked_sub(redeemable_amount_under_management_target_amount)
+                .ok_or(UxdError::MathError)?
+        }
+    };
+
+    let profits_collateral_amount = {
+        let liquidity_collateral_amount_before: u64 =
```

Comment:
nit: no need for all the `before` as we never use the `after` values.

Reply:
Sounds good, changing now!